File: 2020/day06_1.py
# pylint: disable=missing-module-docstring, missing-function-docstring


from typing import Any


def solution(content: str) -> Any:
    return sum(len(set(group.replace("\n", ""))) for group in content.split("\n\n"))


def main() -> None:
    with open("day06.dat", "r", encoding="utf8") as file:
        print(solution(file.read().strip()))


if __name__ == "__main__":
    main()

File: 2020/day06-1.ts
function solution(content: string) {
  return content
    .split("\n\n")
    .map((group) => new Set(group.replaceAll("\n", "")).size)
    .reduce((acc, cur) => acc + cur, 0);
}

Deno.readTextFile("day06.dat")
  .then((content) => content.trim())
  .then(solution)
  .then(console.log);
